### ISSUE TITLE
Text toNumeric needs to handle leading whitespace

### ISSUE THREAD

**Issue by davidhesselbom** [cogneco/ooc-kean#643]
This 
```ooc
text := t" 1.0"
println(text toFloat() toString())
string := text toString()
println(string toFloat() toString())
```
prints
```
0.00
1.00
```
because `String` uses the standard library function `strtof` under the hood, and it can handle leading whitespace, while the conversion functions in `Text` cannot. This has lead to #575 limiting parse functions to work only when the input has no whitespace separating the components of e.g. a `FloatPoint2D`.

The fix is simple: implement a `trim` function in `Text` (could be useful for other reasons, too) and use it on `Text` instances before trying to convert it to a numeric type, or, better yet, let the conversion functions call `trim` before they do anything else.

**Comment by davidhesselbom** [comment]
`strtof` can also handle a leading `+`, for some reason. @sebastianbaginski, do you think it would be a better idea to use `strtof` or to mimic its behavior?

**Comment by sebastianbaginski** [comment]
I vote for implementing it without calling external functions, `trim` can be useful anyway.

**Comment by davidhesselbom** [comment]
Fine by me. Are you up for it?

**Comment by sebastianbaginski** [comment]
Sorry, not at the moment, I still have some PR debts to pay here and in the other project:)

**Comment by davidhesselbom** [comment]
Alright then.

**Comment by thomasfanell** [comment]
I did actually implement trim (left, right and both) for Text quite some time ago. I think I may still have it at home. Otherwise I'll rewrite and PR this evening or tomorrow.

**Comment by thomasfanell** [comment]
Was this fixed by #744?

**Comment by davidhesselbom** [comment]
It was. I think maybe the `toFloat` etc. function should be able to handle leading `+` and maybe even `+ `, but that belongs in another issue, if it turns out we need it.
